### PR TITLE
Update PropTypes to get in sync with React v15.5.4

### PR DESCRIPTION
**Pull request by yamafaktory**
Hi @threepointone,

First of all, thanks for you awesome work, we use glamor in production in our codebase at work and are more than happy with it 👍.

Latest React version (v15.5.0) has extracted the built-in prop types to a separate package https://facebook.github.io/react/blog/2017/04/07/react-v15.5.0.html#migrating-from-react.proptypes. This PR aimed at solving the ugly warning glamor's users will get if they switch to this new version.

### CODE REVIEW

File: examples/tweet/content.js [hunk 1]
```diff
@@ -35,7 +36,7 @@ export class Content extends React.Component {
   }
   render() {
     let { media, text } = this.props
-    let styles = this.styles
```
Comment (yamafaktory) [inline examples/tweet/content.js]: Sorry for the noise, my editor (emacs) auto-magically removes trailing spaces...

Comment (ChristopherBiscardi) [inline examples/tweet/content.js]: Mine does too, I end up using sublime to edit these files, which isn't the greatest :/ I would've suggested using prettier on the entire project but I think the intent is to move glamor's implementation to typescript.

Comment (donaldpipowitch) [inline examples/tweet/content.js]: Prettier should be soon available for TypeScript \o/

File: package.json [hunk 2]
```diff
@@ -82,10 +82,11 @@
     "pegjs": "^0.10.0",
     "preact": "^7.1.0",
     "preact-compat": "^3.9.4",
+    "prop-types": "^15.5.8",
```
Comment (dan-lee) [inline package.json]: Shouldn't `prop-types` be a regular dependency?

Comment (otbe) [inline package.json]: 👍 

(This is what https://github.com/reactjs/prop-types#how-to-depend-on-this-package recommends for libs)

Comment (otbe) [inline package.json]: 👍

Comment (yamafaktory) [inline package.json]: @dan-lee oh, I missed that comment! @otbe done 🚀.

**Comment by yamafaktory** [comment]
@threepointone Hi and sorry pinging you directly (I guess you're quite busy with *Glam* those days 👍 ) but that would be pretty awesome if you can eventually review that PR. Thanks for your awesome work!

**Comment by donaldpipowitch** [comment]
@yamafaktory Thank you!

**Comment by yamafaktory** [comment]
@donaldpipowitch Do you plan to create a new release? Would be awesome by the way!

**Comment by donaldpipowitch** [comment]
@yamafaktory Yeah, can you test 2.20.25? 👍

**Comment by yamafaktory** [comment]
@donaldpipowitch sure! Will report back to you!

**Comment by yamafaktory** [comment]
@donaldpipowitch sorry for the late reply, everything works fine, thanks!

**Comment by donaldpipowitch** [comment]
Awesome. Thank you!

**Pull request merged**